src/parsers/routeToLocation.js

```diff
@@ -19,7 +19,9 @@ const createMatchRouteToPath = registry => ({ id, params = {}, query = {}, hash
     let pathname;
 
     try {
-        pathname = matcher(params, { pretty: true });
+        // path-to-regexp (1.6.0): encodeURI by default, disable it with decodeURI
+        // 'pretty' flag disable all encoding, besides '/', '?', '#'
+        pathname = decodeURIComponent(matcher(params));
```

Comment:
Cause in `path-to-regexp` pretty ? encodeURIComponentPretty : encodeURIComponent
Then `decodeURIComponent(encodeURIComponent(content)) === content`